packages/playwright-crawler/src/internals/adaptive-playwright-crawler.ts

```diff
@@ -315,22 +327,27 @@ export class AdaptivePlaywrightCrawler extends PlaywrightCrawler {
         const $ = load(response.body);
 
         try {
-            await addTimeoutToPromise(
-                async () => this.adaptiveRequestHandler({
-                    request: crawlingContext.request,
-                    log: crawlingContext.log,
-                    querySelector: (selector) => $(selector) as Cheerio<Element>,
-                    enqueueLinks: async (options: Parameters<RestrictedCrawlingContext['enqueueLinks']>[0] = {}) => {
-                        const urls = extractUrlsFromCheerio($, options.selector, options.baseUrl ?? loadedUrl);
-                        await result.enqueueLinks({ ...options, urls });
-                    },
-                    addRequests: result.addRequests,
-                    pushData: result.pushData,
-                    useState: result.useState,
-                    getKeyValueStore: result.getKeyValueStore,
-                }),
-                this.requestHandlerTimeoutInnerMillis,
-                'Request handler timed out',
+            await withCheckedStorageAccess(
+                () => {
+                    throw new Error('Directly accessing storage in a request handler is not allowed');
+                },
+                async () => addTimeoutToPromise(
```

Comment:
I hope this wont break the entirety of the universe, since `addTimeoutToPromise` also uses ALS internally :D 

Did some local testing and it looks good. Which reminds me we should add the adaptive crawler e2e test.

Reply:
It's a different storage, so it should be good. The issue for adaptive e2e is here - https://github.com/apify/crawlee/issues/2350